core/src/com/po/fuck/AssetsManagement/WeaponSpriteManager.java

```diff
@@ -0,0 +1,9 @@
+package com.po.fuck.AssetsManagement;
+
+import com.po.fuck.model.Sprites.WeaponSpriteInfo;
+
+public class WeaponSpriteManager extends BasicSpriteManager {
+    public <T> WeaponSpriteInfo getWeaponSpriteInfo(Class<T> name) {
```

Comment:
Do you know what is inheritance ???
Rename `getBasicSpriteInfo` into `getSpriteInfo`/`getInfo`, and override it here!!!